core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -345,6 +350,55 @@ oppia.directive('questionPlayer', [
             }
           };
 
+          ctrl.reviewConceptCardAndRetryTest = function() {
+            if (!ctrl.failedSkillIds || ctrl.failedSkillIds.length === 0) {
+              throw Error('No failed skills');
+            }
+
+            var failedSkills = [];
+            ctrl.failedSkillIds.forEach(function(skillId) {
+              failedSkills.push(
+                ctrl.scorePerSkillMapping[skillId].description);
+            });
+
+            $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/components/question-directives/question-player/' +
+                'review-and-retry-modal.template.html'
+              ),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance', '$window',
+                'UrlService',
+                function(
+                    $scope, $uibModalInstance, $window,
+                    UrlService) {
+                  $scope.failedSkillIds = ctrl.failedSkillIds;
+                  $scope.failedSkills = failedSkills;
+                  $scope.index = 0;
+                  $scope.currentSkill = $scope.failedSkills[$scope.index];
+
+                  $scope.isLastConceptCard = function() {
+                    return $scope.index === $scope.failedSkills.length - 1;
+                  };
+
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                  };
+
+                  $scope.next = function() {
```

Comment:
Change function name to: goToNextConceptCard()

Reply:
Done.